### DISCUSSION TITLE
Pros and cons of knitr engine for Python/Julia (vs Jupyter)

### DISCUSSION THREAD

**Discussion by baptiste** [quarto-dev/quarto-cli#5162] [Q&A]
I'm realising that I don't understand the status of the (many) languages supported by knitr, in the context of Quarto documents. 

The documentation suggests to use `engine: knitr` for R code, and `engine: jupyter` for Python or Julia, but I'm wondering what would be the pros and cons of using knitr instead with Julia or Python code cells? The following basic example does seem to work, 

````
---
title: "Plots Demo"
author: "Norah Jones"
date: "5/22/2021"
format:
  html:
    code-fold: true
engine: knitr
---

## Parametric Plots

Plot function pair (x(u), y(u)). 
See @fig-parametric for an example.

```{julia}
#| label: fig-parametric
#| fig-cap: "Parametric Plots"
using Plots

plot(sin, 
     x->sin(2x), 
     0, 
     2π, 
     leg=false, 
     fill=(0,:lavender))
```

````

It might seem pointless in this example, but using knitr has the nice side-effect that one can access some features not yet available in the jupyter counterpart. For example, I can access the Yaml metadata, and use it to dynamically change the output of a Julia code chunk, which I think is not currently possible with the Jupyter route (I don't know of a way to access the metadata).

````
---
title: "Plots Demo"
author: "Norah Jones"
date: "5/22/2021"
format:
  html:
    code-fold: true
engine: knitr
draft: false
---

```{r}
if(rmarkdown::metadata$draft){
  .conditional <- "THIS IS A DRAFT"
} else {
  .conditional <- "THIS IS A STORM"
}
library(JuliaCall)
 julia_assign("my_title", .conditional)
```

```{julia}
#| label: fig-parametric
using Plots

plot(sin, 
     x->sin(2x), 
     0, 
     2π, 
     leg=false, 
     fill=(0,:lavender), title=my_title)
```
````

![Screenshot 2023-04-12 at 19 58 28](https://user-images.githubusercontent.com/18970/231391603-e111d3d0-1245-40b5-870d-ff763ac2a59d.png)

Is the difference between knitr and jupyter documented somewhere? For those of us switching from Rmarkdown it would be quite helpful.

**Comment by jangsean** [comment]
I have the same question. If you come to a conclusion, please update. Thx
